addons/phone_validation/models/phone_blacklist.py

```diff
@@ -65,11 +65,20 @@ def _search(self, args, offset=0, limit=None, order=None, count=False, access_ri
         if args:
             new_args = []
             for arg in args:
-                if isinstance(arg, (list, tuple)) and arg[0] == 'number' and isinstance(arg[2], str):
-                    number = arg[2]
-                    sanitized = phone_validation.phone_sanitize_numbers_w_record([number], self.env.user)[number]['sanitized']
-                    if sanitized:
-                        new_args.append([arg[0], arg[1], sanitized])
+                if isinstance(arg, (list, tuple)) and arg[0] == 'number':
+                    if isinstance(arg[2], str):
+                        number = arg[2]
+                        sanitized = phone_validation.phone_sanitize_numbers_w_record([number], self.env.user)[number]['sanitized']
+                        if sanitized:
+                            new_args.append([arg[0], arg[1], sanitized])
+                        else:
+                            new_args.append(arg)
+                    elif isinstance(arg[2], list) and all(isinstance(number, str) for number in arg[2]):
+                        new_list = []
+                        for number in arg[2]:
+                            sanitized = phone_validation.phone_sanitize_numbers_w_record([number], self.env.user)[number]['sanitized']
+                            new_list.append(sanitized or number)
+                        new_args.append([arg[0], arg[1], new_list])
```

Comment:
Instead of append everytime in new_args, we could probably just compute a new "search term" and make the final arg in the end, like
```suggestion
                    if isinstance(arg[2], str):
                        number = arg[2]
                        sanitized = phone_validation.phone_sanitize_numbers_w_record([number], self.env.user)[number]['sanitized']
                        search_term = sanitized or number
                    elif isinstance(arg[2], list) and all(isinstance(number, str) for number in arg[2]):
                        search_term = [
                            phone_validation.phone_sanitize_numbers_w_record([number], self.env.user)[number]['sanitized'] or number
                            for number in arg[2]
                        ]
                    else:
                        search_term = arg[2]
                    new_args.append([arg[0], arg[1], search_term])
```

Reply:
Done